riscv/src/runtime.rs

```diff
@@ -276,39 +276,76 @@ impl Runtime {
         self.syscalls.contains_key(&s)
     }
 
-    pub fn with_poseidon(mut self) -> Self {
```

Comment:
since we have the `*_continuations` versions of this method, id make this one private

Reply:
update